dotnet/generate-vs-workload.csharp

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env /Library/Frameworks/Mono.framework/Commands/csharp -s
+
+// arguments are:
+//   --shorten long=short
+//   --platform <platform> <version>
+//   --windows-platform <platform>
+
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+using System.Xml;
+
+var shorten = new List<(string, string)> ();
+var platforms = new List<(string, string)> ();
+var windowsPlatforms = new List<string> ();
+var outputPath = string.Empty;
+
+var args = new Queue<string> (Environment.GetCommandLineArgs ());
+
+// Skip first three arguments:
+string discard; // store the return value from Dequeue somewhere, otherwise it ends up being printed to stdout.
+discard = args.Dequeue (); // <executable>
+discard = args.Dequeue (); // -s
+discard = args.Dequeue (); // <script>
+GC.KeepAlive (discard); // avoid compiler warning
+
+while (args.Any ()) {
+	var arg = args.Dequeue ();
+	switch (arg) {
+	case "--shorten":
+		var values = args.Dequeue ().Split ('=');
+		shorten.Add ((values [0], values [1]));
+		break;
+	case "--platform":
+		var platform = args.Dequeue ();
+		var version = args.Dequeue ();
+		platforms.Add ((platform, version));
+		break;
+	case "--windows-platform":
+		windowsPlatforms.Add (args.Dequeue ());
+		break;
+	case "--output":
+		outputPath = args.Dequeue ();
+		break;
+	default:
+		Console.Error.WriteLine ($"Unknown argument: {arg}");
+		Environment.Exit (1);
+		break;
+	}
+}
+
+using (TextWriter writer = new StreamWriter (outputPath)) {
+	writer.WriteLine ($"<?xml version=\"1.0\" encoding=\"utf-8\"?>");
+	writer.WriteLine ($"<Project>");
+	writer.WriteLine ($"  <PropertyGroup>");
+	writer.WriteLine ($"    <TargetName>Microsoft.NET.Sdk.MaciOS.Workload</TargetName>");
```

Comment:
We will need to also partially generate `<TargetName>` to ensure that we get unique VS manifest file names for each branch that makes up a complete VS insertion. We could optionally add a suffix to this for any special branches, perhaps something like `Microsoft.NET.Sdk.MaciOS.Workload.Xcode14`.

Reply:
@pjcollins would it make sense to create one file per platform?

Reply:
Doing one vsman file per platform would require changes in yaml-templates but could be an option. Another option that wouldn't require changes in yaml-templates would be to name the VSMan file after the platforms that will be included (something like `Microsoft.NET.Sdk.tvOS.iOS.Workload` maybe).

Reply:
@pjcollins is the name included in a path somewhere? i.e. could we run into path length problems if we name it something like `Microsoft.NET.Sdk.iOS.tvOS.MacCatalyst.macOS.Workload`?

Reply:
Also is the `ManifestBuildVersion` field used for something important? Because we're right now using the iOS version, and if iOS isn't included, whatever the next platform happens to be, which seems rather unreliable/unpredictable.

Reply:
I've updated the code to add all the enabled platforms to the TargetName. Resulting file is now: https://gist.github.com/rolfbjarne/9e3be346ef1bb1823c96dca8162624b5

Reply:
I do not think we need to be concerned with path length wrt the vsman file name. The `ManifestBuildVersion` property is unimportant, it just needs to be set. If we care about it being more predictable we could set it to the version of the first Apple platform included in the vsman file.